### ISSUE TITLE
LanguageResources installed from Plugins are saved incorectly cased

### ISSUE THREAD

**Issue by manuel3108** [nopSolutions/nopCommerce#5942]
nopCommerce version: 4.40.4

Steps to reproduce the problem: 
* Install a plugin with the following in it's Install() method (Make sure you pass a dictionary):
```c#
await _localizationService.AddLocaleResourceAsync(new Dictionary<string, string>
 {
     ["Plugins.Shipping.FixedByWeightByTotal.AddRecord"] = "Add record",
}
```
* After the installation, have a look at your db, and you will see in the table `LocaleStringResource` a `ResourceName` with the following casing: `Plugins.Shipping.FixedByWeightByTotal.AddRecord`
* If you try to access this value from the NopBackend, Nop will try to access the resource name in lowercase, but will be unable to find it, because of wrong casing. This is one way I was trying to access it:
```html
<nop-label asp-for="AddRecord" />
```

Expected:
All language tokens are lowercase before insertion into database.

Problem:
The problem is probably around here: https://github.com/nopSolutions/nopCommerce/blob/develop/src/Libraries/Nop.Services/Localization/LocalizationService.cs#L733-L754

Note:
Can also be tested with installing the FixedByWeightByTotal plugin into a fresh nop4.40.4 DB.

**Comment by manuel3108** [comment]
Apparently also happens when inserting a new custom string via the UI on the languages site: /Admin/Language/Edit/1

**Comment by skoshelev** [comment]
Closed #5942

**Comment by skoshelev** [comment]
Closed #5942
